hnn_core/network.py

```diff
@@ -1127,42 +1128,55 @@ def _instantiate_drives(self, tstop, n_trials=1):
                 self.external_drives[
                     drive['name']]['events'].append(event_times)
 
-    def add_tonic_bias(self, *, cell_type=None, amplitude=None,
-                       t0=None, tstop=None):
-        """Attach parameters of tonic biasing input for a given cell type.
+    def add_tonic_bias(self, *, cell_type=None, amplitude, t0=0, tstop=None):
+        """Attaches parameters of tonic bias input for given cell types
 
         Parameters
         ----------
-        cell_type : str
-            The cell type whose cells will get the tonic input.
-            Valid inputs are those in `net.cell_types`.
-        amplitude : float
-            The amplitude of the input.
+        cell_types : str | None
+            The name of the cell type to add a tonic input. When supplied,
+            a float value must be provided with the `amplitude` keyword.
+            Valid inputs are those listed in  `net.cell_types`.
+        amplitude: dict | float
+            A dictionary of cell type keys (str) to amplitude values (float).
+            Valid inputs for cell types are those listed in `net.cell_types`.
+            If `cell_types` is not None, `amplitude` should be
+            a float indicating the amplitude of the tonic input
+            for the specified cell type.
         t0 : float
             The start time of tonic input (in ms). Default: 0 (beginning of
-            simulation).
+            simulation). This value will be applied to all the  tonic biases if
+            multiple are specified with the `amplitude` keyword.
         tstop : float
             The end time of tonic input (in ms). Default: end of simulation.
+            This value will be applied to all the  tonic biases if
+            multiple are specified with the `amplitude` keyword.
         """
-        if (cell_type is None or amplitude is None):
-            raise ValueError('cell_type and amplitude must be defined'
-                             f', got {cell_type}, {amplitude}')
-        if 'tonic' not in self.external_biases:
-            self.external_biases['tonic'] = dict()
-        if cell_type in self.external_biases['tonic']:
-            raise ValueError(f'Tonic bias already defined for {cell_type}')
-        if t0 is None:
-            t0 = 0
-        if cell_type not in self.cell_types:
-            raise ValueError(f'cell_type must be one of '
-                             f'{list(self.cell_types.keys())}. '
-                             f'Got {cell_type}')
 
-        self.external_biases['tonic'][cell_type] = {
-            'amplitude': amplitude,
-            't0': t0,
-            'tstop': tstop
-        }
+        # old functionality single cell type - amplitude
+        if cell_type is not None:
+            warnings.warn('cell_type argument will be deprecated and '
+                          'removed in future releases. Use amplitude as a '
+                          'cell_type:str,amplitude:float dictionary.'
```

Comment:
probably a bit late but I find this a little cryptic for the user. Would have instead said:

'dictionary to provide cell-specific tonic bias'